python/ray/tempfile_services.py

```diff
@@ -0,0 +1,261 @@
+import binascii
+import collections
+import datetime
+import errno
+import logging
+import os
+import shutil
+import tempfile
+
+import ray.utils
+
+logger = logging.getLogger(__name__)
+
+RAY_TEMPDIR = 'RAY_TEMPDIR'
+RAY_OBJECT_STORE_SOCKET_NAME = 'RAY_OBJECT_STORE_SOCKET_NAME'
+
+_incremental_dict = collections.defaultdict(lambda :0)
+
+
+def make_inc_temp(suffix="", prefix="", dir=None):
```

Comment:
`dir` is a Python function, so let's use a different name, e.g., `directory_name`